In [9]:
conn = sqlite3.connect(db)
#creating a cursor object
cur = conn.cursor()

In [3]:
#numpy ->To perform the mathmatical operations

import numpy as np 
#pandas -> data analysis -> DE, DM ,DC
import pandas as pd 
#matplotlib ->Daat visualisation tool
import matplotlib.pyplot as plt 
#seaborn - Data visulisation tool
import seaborn as snsl
#regex -> data cleaning 
import re
#sqlite-> server less database
import sqlite3

#planning /approach

STEP1: connect database sqlit3.connect(database_name)
STEP2: Use the cursor function - Database_variable.cursor

workflow

1. I will establish a connection to the SQLITE Database by creating a connection object
2. I will create a cursor object using the cursor() method
3. I will execute the query cursor_variable.execute('query')
4. Fetch the data then use fetchall() method of the cursor variable/object
5. I will create a DataFrame w.r.t the SQlite
6. Data Analysis - Data Manipulation, Data Exploration, Data Cleaning, Data Visualization
7. I will write a conclusion at every step

In [7]:

#storing the path of the database into a vaiable "db"
db = "/content/drive/MyDrive/movies.sqlite"
#creating a database connection object 
conn = sqlite3.connect(db)
#creating a cursor object
cur = conn.cursor()

In [11]:
#fetch all the data of the movies from the movies table in the above database 
cur.execute("select * from movies")
#creating a cursor object
movies =cur.fetchall()

In [15]:
#creating a dataframe 
columns = ["id","original_title","budget","popularity","releasedate","revenue","title",
                 "vote","vote_average","vote_count","overview","uid","director_id"]
movies_df = pd.DataFrame(movies,columns=columns)
movies_df 

id                            original_title     budget  popularity  \
0     43597                                    Avatar  237000000         150   
1     43598  Pirates of the Caribbean: At World's End  300000000         139   
2     43599                                   Spectre  245000000         107   
3     43600                     The Dark Knight Rises  250000000         112   
4     43601                               John Carter  260000000          43   
...     ...                                       ...        ...         ...   
4768  48395                               El Mariachi     220000          14   
4769  48396                                 Newlyweds       9000           0   
4770  48397                 Signed, Sealed, Delivered          0           1   
4771  48398                          Shanghai Calling          0           0   
4772  48399                         My Date with Drew          0           1   

     releasedate     revenue                                     title  vote  \
0     2009-12-10  2787965087                                    Avatar   7.2   
1     2007-05-19   961000000  Pirates of the Caribbean: At World's End   6.9   
2     2015-10-26   880674609                                   Spectre   6.3   
3     2012-07-16  1084939099                     The Dark Knight Rises   7.6   
4     2012-03-07   284139100                               John Carter   6.1   
...          ...         ...                                       ...   ...   
4768  1992-09-04     2040920                               El Mariachi   6.6   
4769  2011-12-26           0                                 Newlyweds   5.9   
4770  2013-10-13           0                 Signed, Sealed, Delivered   7.0   
4771  2012-05-03           0                          Shanghai Calling   5.7   
4772  2005-08-05           0                         My Date with Drew   6.3   

      vote_average                                         vote_count  \
0            11800  In the 22nd century, a paraplegic Marine is di...   
1             4500  Captain Barbossa, long believed to be dead, ha...   
2             4466  A cryptic message from Bond’s past sends him o...   
3             9106  Following the death of District Attorney Harve...   
4             2124  John Carter is a war-weary, former military ca...   
...            ...                                                ...   
4768           238  El Mariachi just wants to play his guitar and ...   
4769             5  A newlywed couple's honeymoon is upended by th...   
4770             6  "Signed, Sealed, Delivered" introduces a dedic...   
4771             7  When ambitious New York attorney Sam is sent t...   
4772            16  Ever since the second grade when he first saw ...   

                                               overview     uid  director_id  
0                           Enter the World of Pandora.   19995         4762  
1        At the end of the world, the adventure begins.     285         4763  
2                                 A Plan No One Escapes  206647         4764  
3                                       The Legend Ends   49026         4765  
4                  Lost in our world, found in another.   49529         4766  
...                                                 ...     ...          ...  
4768  He didn't come looking for trouble, but troubl...    9367         5097  
4769  A newlywed couple's honeymoon is upended by th...   72766         6485  
4770                                               None  231617         7108  
4771                           A New Yorker in Shanghai  126186         7109  
4772                                               None   25975         7110  

[4773 rows x 13 columns]

In [16]:
#display all the DataFrame
movies_df

id                            original_title     budget  popularity  \
0     43597                                    Avatar  237000000         150   
1     43598  Pirates of the Caribbean: At World's End  300000000         139   
2     43599                                   Spectre  245000000         107   
3     43600                     The Dark Knight Rises  250000000         112   
4     43601                               John Carter  260000000          43   
...     ...                                       ...        ...         ...   
4768  48395                               El Mariachi     220000          14   
4769  48396                                 Newlyweds       9000           0   
4770  48397                 Signed, Sealed, Delivered          0           1   
4771  48398                          Shanghai Calling          0           0   
4772  48399                         My Date with Drew          0           1   

     releasedate     revenue                                     title  vote  \
0     2009-12-10  2787965087                                    Avatar   7.2   
1     2007-05-19   961000000  Pirates of the Caribbean: At World's End   6.9   
2     2015-10-26   880674609                                   Spectre   6.3   
3     2012-07-16  1084939099                     The Dark Knight Rises   7.6   
4     2012-03-07   284139100                               John Carter   6.1   
...          ...         ...                                       ...   ...   
4768  1992-09-04     2040920                               El Mariachi   6.6   
4769  2011-12-26           0                                 Newlyweds   5.9   
4770  2013-10-13           0                 Signed, Sealed, Delivered   7.0   
4771  2012-05-03           0                          Shanghai Calling   5.7   
4772  2005-08-05           0                         My Date with Drew   6.3   

      vote_average                                         vote_count  \
0            11800  In the 22nd century, a paraplegic Marine is di...   
1             4500  Captain Barbossa, long believed to be dead, ha...   
2             4466  A cryptic message from Bond’s past sends him o...   
3             9106  Following the death of District Attorney Harve...   
4             2124  John Carter is a war-weary, former military ca...   
...            ...                                                ...   
4768           238  El Mariachi just wants to play his guitar and ...   
4769             5  A newlywed couple's honeymoon is upended by th...   
4770             6  "Signed, Sealed, Delivered" introduces a dedic...   
4771             7  When ambitious New York attorney Sam is sent t...   
4772            16  Ever since the second grade when he first saw ...   

                                               overview     uid  director_id  
0                           Enter the World of Pandora.   19995         4762  
1        At the end of the world, the adventure begins.     285         4763  
2                                 A Plan No One Escapes  206647         4764  
3                                       The Legend Ends   49026         4765  
4                  Lost in our world, found in another.   49529         4766  
...                                                 ...     ...          ...  
4768  He didn't come looking for trouble, but troubl...    9367         5097  
4769  A newlywed couple's honeymoon is upended by th...   72766         6485  
4770                                               None  231617         7108  
4771                           A New Yorker in Shanghai  126186         7109  
4772                                               None   25975         7110  

[4773 rows x 13 columns]

In [17]:

movies_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4773 entries, 0 to 4772
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   id              4773 non-null   int64  
 1   original_title  4773 non-null   object 
 2   budget          4773 non-null   int64  
 3   popularity      4773 non-null   int64  
 4   releasedate     4773 non-null   object 
 5   revenue         4773 non-null   int64  
 6   title           4773 non-null   object 
 7   vote            4773 non-null   float64
 8   vote_average    4773 non-null   int64  
 9   vote_count      4770 non-null   object 
 10  overview        3951 non-null   object 
 11  uid             4773 non-null   int64  
 12  director_id     4773 non-null   int64  
dtypes: float64(1), int64(7), object(5)
memory usage: 484.9+ KB


In [23]:
#fetch all the data of the director from the director table in the above database 
cur.execute("select * from directors")
#creating a cursor object
directors=cur.fetchall()

In [30]:
directors_df

name  director_id  gender      uid department
0         James Cameron         4762       2     2710  Directing
1        Gore Verbinski         4763       2     1704  Directing
2            Sam Mendes         4764       2       39  Directing
3     Christopher Nolan         4765       2      525  Directing
4        Andrew Stanton         4766       2        7  Directing
...                 ...          ...     ...      ...        ...
2344      Shane Carruth         7106       2    76624  Directing
2345   Neill Dela Llana         7107       0  1174437  Directing
2346        Scott Smith         7108       0  1219158  Directing
2347        Daniel Hsia         7109       2   208138  Directing
2348   Brian Herzlinger         7110       2    85563  Directing

[2349 rows x 5 columns]

In [28]:
#creating a DataFrame
directors_df = pd.DataFrame(directors , columns=["name","director_id","gender","uid","department"])

In [29]:
directors_df

name  director_id  gender      uid department
0         James Cameron         4762       2     2710  Directing
1        Gore Verbinski         4763       2     1704  Directing
2            Sam Mendes         4764       2       39  Directing
3     Christopher Nolan         4765       2      525  Directing
4        Andrew Stanton         4766       2        7  Directing
...                 ...          ...     ...      ...        ...
2344      Shane Carruth         7106       2    76624  Directing
2345   Neill Dela Llana         7107       0  1174437  Directing
2346        Scott Smith         7108       0  1219158  Directing
2347        Daniel Hsia         7109       2   208138  Directing
2348   Brian Herzlinger         7110       2    85563  Directing

[2349 rows x 5 columns]

In [31]:
directors_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2349 entries, 0 to 2348
Data columns (total 5 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   name         2349 non-null   object
 1   director_id  2349 non-null   int64 
 2   gender       2349 non-null   int64 
 3   uid          2349 non-null   int64 
 4   department   2349 non-null   object
dtypes: int64(3), object(2)
memory usage: 91.9+ KB


In [33]:
movies_df.shape

(4773, 13)

In [34]:
directors_df.shape

(2349, 5)

In [35]:
movies_df.describe()

id        budget   popularity       revenue         vote  \
count   4773.000000  4.773000e+03  4773.000000  4.773000e+03  4773.000000   
mean   45985.680913  2.922461e+07    21.136602  8.277710e+07     6.109177   
std     1382.032760  4.078679e+07    31.864132  1.632374e+08     1.146789   
min    43597.000000  0.000000e+00     0.000000  0.000000e+00     0.000000   
25%    44790.000000  9.500000e+05     4.000000  0.000000e+00     5.600000   
50%    45983.000000  1.500000e+07    13.000000  1.947811e+07     6.200000   
75%    47176.000000  4.000000e+07    28.000000  9.361701e+07     6.800000   
max    48399.000000  3.800000e+08   875.000000  2.787965e+09    10.000000   

       vote_average            uid  director_id  
count   4773.000000    4773.000000  4773.000000  
mean     694.541588   56244.396396  5586.661429  
std     1237.251355   87452.149183   670.413263  
min        0.000000       5.000000  4762.000000  
25%       55.000000    8998.000000  5011.000000  
50%      238.000000   14517.000000  5389.000000  
75%      742.000000   57876.000000  6086.000000  
max    13752.000000  459488.000000  7110.000000

#how many movies are present in the movies table

In [36]:
movies_df.columns

Index(['id', 'original_title', 'budget', 'popularity', 'releasedate',
       'revenue', 'title', 'vote', 'vote_average', 'vote_count', 'overview',
       'uid', 'director_id'],
      dtype='object')

In [39]:
cur.execute("select count(id) from movies")
count = cur.fetchall()
print(f"the number of movies present in the movies table is {count[0][0]}")

the number of movies present in the movies table is 4773


In [40]:
directors_df.columns

Index(['name', 'director_id', 'gender', 'uid', 'department'], dtype='object')

Q2. find these 3 directors : james cameron,luc  besson , john woo

In [44]:
#select * from directors where name in ('james cameron' ,'luc besson','john woo')
cur.execute("select*from directors where name='james cameron' or name='luc besson' or name='john woo'" )
three_directors = cur.fetchall()
print(f"here are the info for the asked directors{three_directors}")

here are the info for the asked directors[]


Q3.find all the directors with the name starting with 'steven' 

In [46]:
cur.execute("select * from directors where name like 'steven' ")
name_like = cur.fetchall()
print(f"these are all directors whohsse name starts with 'steven'{name_like}")

these are all directors whohsse name starts with 'steven'[]


#count all the female directors

In [53]:
cur.execute( "select Count(name) from directors where gender =1")
female_directors = cur.fetchall()
print(f"the number of female  directors is {female_directors[0][0]}")

the number of female  directors is 150


#conclusion
the number of female director  is 150

Q5.find the name of the 10th first female director 

In [70]:
cur.execute("select name  from directors where gender =1")
tenth_female = cur.fetchall()
print(f"the tenth_female_director is{tenth_female[0][0]}")

the tenth_female_director isBrenda Chapman


Q6.find the top three most popular movies 

In [77]:
cur.execute("select original_title from movies order by popularity desc limit 3 ")
most_popular=cur.fetchall()
print(f"the top 3 most popular are{most_popular[0][0]}")

the top 3 most popular areMinions


Q7.top3 most bankable budget

In [80]:
cur.execute("select original_title from movies order by budget desc limit 3 ")
most_bankable=cur.fetchall()
print(f"the top 3 most bankable are{most_bankable}")

the top 3 most bankable are[('Pirates of the Caribbean: On Stranger Tides',), ("Pirates of the Caribbean: At World's End",), ('Avengers: Age of Ultron',)]


Q8. most awarded average rated movie since jan 1 st 2000

In [82]:
cur.execute("select original_title from movies where release_date >2000-01-01 order by vote_average desc limit 1 ")
most_awarded=cur.fetchall()
print(f"the top 3 most popular are{most_awarded[0][0]}")

the top 3 most popular areStiff Upper Lips


Q9 find all the movie that has been directed by brenda chapman

In [87]:
query = """select original_title from movies  left join directors on movies.director_id '
cur.excute("select original_title from movies where release_date >2000-01-01 order by vote_average desc limit 1 ")
directed_movie = cur.fetchall()

print(f"the movies directed by brenda chapman is /are {directed_movie[0][0]}")
"""

Q10. find out the director who has made the most movies

In [89]:
movies_df.columns

Index(['id', 'original_title', 'budget', 'popularity', 'releasedate',
       'revenue', 'title', 'vote', 'vote_average', 'vote_count', 'overview',
       'uid', 'director_id'],
      dtype='object')

In [92]:
directors_df.columns

Index(['name', 'director_id', 'gender', 'uid', 'department'], dtype='object')

In [104]:
query = "select name from directors join movies on directors.id=director_id GROUP BY director_id order by  count(name) desc limit 1"
cur.execute(query)
directed_movie = cur.fetchall()
print(f"the movies directed by brenda chapman is / are {directed_movie[0][0]}")

the movies directed by brenda chapman is / are Steven Spielberg


Q11. find out the director who is most bankable 

In [111]:
query = "select name from directors join movies on directors.id= director_id GROUP BY director_id order by  count(name) desc limit 1"
cur.execute(query)
bankable_movie = cur.fetchall()
print(f"the movies bankable director is {bankable_movie[0][0]}")

the movies bankable director is Steven Spielberg


Q12. find out the top 10 highest  buduget making  movies

In [114]:
query = "select original_title from movies order by budget desc limit 10"
cur.execute(query)
top_10 =cur.fetchall()
print(f"the top 10 highest budget movies are {top_10}")


the top 10 highest budget movies are [('Pirates of the Caribbean: On Stranger Tides',), ("Pirates of the Caribbean: At World's End",), ('Avengers: Age of Ultron',), ('Superman Returns',), ('John Carter',), ('Tangled',), ('Spider-Man 3',), ('The Lone Ranger',), ('The Dark Knight Rises',), ('Harry Potter and the Half-Blood Prince',)]


#Q13. find the top 10 revenue making movies

In [115]:
movies_df.columns

Index(['id', 'original_title', 'budget', 'popularity', 'releasedate',
       'revenue', 'title', 'vote', 'vote_average', 'vote_count', 'overview',
       'uid', 'director_id'],
      dtype='object')

In [117]:
query = "select original_title  from movies order by revenue  desc limit 10"
cur.execute(query)
top_10_revenue =cur.fetchall()
print(f"the top 10 revenue making  movies are {top_10_revenue}")

the top 10 revenue making  movies are [('Avatar',), ('Titanic',), ('The Avengers',), ('Jurassic World',), ('Furious 7',), ('Avengers: Age of Ultron',), ('Frozen',), ('Iron Man 3',), ('Minions',), ('Captain America: Civil War',)]


Q14.find out the top 10 most popular movies with the highest rating(vote_avg)

In [118]:
movies_df.columns

Index(['id', 'original_title', 'budget', 'popularity', 'releasedate',
       'revenue', 'title', 'vote', 'vote_average', 'vote_count', 'overview',
       'uid', 'director_id'],
      dtype='object')

In [121]:
query = "select original_title  from movies order by popularity,vote_average desc limit 10"
cur.execute(query)
top_10_rating =cur.fetchall()
print(f"the top 10 most popular w.r.t. the highest rating are {top_10_rating}")

the top 10 most popular w.r.t. the highest rating are [('Stiff Upper Lips',), ('Dancer, Texas Pop. 81',), ('Sardaarji',), ("One Man's Hero",), ('There Goes My Baby',), ('Counting',), ('Déjà Vu',), ('Dolphins and Whales: Tribes of the Ocean',), ('Lake of Fire',), ('Dreaming of Joseph Lees',)]
